src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -278,13 +297,82 @@ private bool DynamicDataAttached(IDictionary<string, object?> sourceLevelParamet
+            }
+
+            if (testMethodInfo.Parent.Parent.AssemblyCleanupMethod is not null)
+            {
+                tests.Add(GetAssemblyFixtureTest(testMethodInfo.Parent.Parent.AssemblyCleanupMethod, assemblyName, className, classFullName, assemblyLocation, Constants.AssemblyCleanupFixtureTrait));
+            }
+        }
+
+        if (!fixtureTests.Contains(assemblyLocation + classFullName))
+        {
+            _ = fixtureTests.Add(assemblyLocation + classFullName);
+
+            if (testMethodInfo.Parent.ClassInitializeMethod is not null)
+            {
+                tests.Add(GetClassFixtureTest(testMethodInfo.Parent.ClassInitializeMethod, assemblyName, className, classFullName, assemblyLocation, Constants.ClassInitializeFixtureTrait));
+            }
+
+            if (testMethodInfo.Parent.ClassCleanupMethod is not null)
+            {
+                tests.Add(GetClassFixtureTest(testMethodInfo.Parent.ClassCleanupMethod, assemblyName, className, classFullName, assemblyLocation, Constants.ClassCleanupFixtureTrait));
+            }
+        }
+
+        static UnitTestElement GetAssemblyFixtureTest(MethodInfo methodInfo, string assemblyName, string className, string classFullName,
+            string assemblyLocation, string fixtureType)
+        {
+            string methodName = GetMethodName(methodInfo);
+            string[] hierarchy = [null!, assemblyName, $"[{Constants.AssemblyFixturesHierarchyName}]", methodName];
+            return GetFixtureTest(classFullName, assemblyLocation, fixtureType, methodName, hierarchy);
+        }
+
+        static UnitTestElement GetClassFixtureTest(MethodInfo methodInfo, string assemblyName, string className, string classFullName,
+            string assemblyLocation, string fixtureType)
+        {
+            string methodName = GetMethodName(methodInfo);
+            string[] hierarchy = [null!, classFullName, methodName];
+            return GetFixtureTest(classFullName, assemblyLocation, fixtureType, methodName, hierarchy);
+        }
+
+        static string GetMethodName(MethodInfo methodInfo)
+        {
+            ParameterInfo[] args = methodInfo.GetParameters();
+            return args.Length > 0
+                ? $"{methodInfo.Name}({string.Join(",", args.Select(a => a.ParameterType.FullName))})"
+                : methodInfo.Name;
+        }
+
+        static UnitTestElement GetFixtureTest(string classFullName, string assemblyLocation, string methodType, string methodName, string[] hierarchy)
+        {
+            var method = new TestMethod(classFullName, methodName,
+                hierarchy, methodName, classFullName, assemblyLocation, false,
+                TestIdGenerationStrategy.FullyQualified);
+            return new UnitTestElement(method)
+            {
+                DisplayName = $"[{methodType}].{methodName}",
```

Comment:
I would prefer not to have the parameters for the display name

Reply:
We need to populate full methodName for navigation from TE to code. So there will be one more string to remove parameters.